### PR TITLE
neovim: link jemalloc statically

### PR DESCRIPTION
**Pull request by ilovezfs**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

See https://github.com/neovim/neovim/pull/6979.

### CODE REVIEW

**Comment by 0xABAB** [comment]
Relevant: https://github.com/NixOS/nixpkgs/issues/27223#issuecomment-313788342.

TL;DR Just because someone asks you to merge something, doesn't mean that you should.

**Comment by ilovezfs** [comment]
@0xABAB are you going to fix neovim?
